cmd/sonobuoy/app/results.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2018 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/heptio/sonobuoy/pkg/client/results"
+	"github.com/heptio/sonobuoy/pkg/errlog"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type resultsInput struct {
+	archive  string
+	plugin   string
+	detailed bool
+}
+
+func NewCmdResults() *cobra.Command {
+	data := resultsInput{}
+	cmd := &cobra.Command{
+		Use:   "results archive.tar.gz",
+		Short: "Inspect e2e test results. Optionally rerun failed tests",
+		Run: func(cmd *cobra.Command, args []string) {
+			data.archive = args[0]
+			if err := result(data); err != nil {
+				errlog.LogError(errors.Wrapf(err, "could not open sonobuoy archive: %v", args[0]))
+				os.Exit(1)
+			}
+		},
+		Args: cobra.ExactArgs(1),
+	}
+	AddPluginFlag(&data.plugin, cmd.Flags())
+	cmd.Flags().BoolVar(
+		&data.detailed, "detailed", false,
+		"Prints more detailed information regarding the plugin if available",
+	)
+
+	return cmd
+}
+
+func getReader(filepath string) (*results.Reader, func(), error) {
+	f, err := os.Open(filepath)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "could not open sonobuoy archive: %v", filepath)
+	}
+
+	gzr, err := gzip.NewReader(f)
+	if err != nil {
+		return nil, func() { f.Close() }, errors.Wrap(err, "could not make a gzip reader")
+	}
+
+	r := results.NewReaderWithVersion(gzr, results.VersionTen)
+	return r, func() { gzr.Close(); f.Close() }, nil
+}
+
+func result(input resultsInput) error {
+	r, cleanup, err := getReader(input.archive)
+	defer cleanup()
+	if err != nil {
+		errlog.LogError(err)
+		os.Exit(1)
+	}
+
+	obj, err := r.PluginResults(input.plugin)
+	if input.detailed {
+		printResultsDetails("", obj, input)
+	} else {
+		printResultsSummary(*obj)
+	}
+	return nil
+}
+
+func printResultsDetails(prefix string, o *results.Item, input resultsInput) {
```

Comment:
If you wanted to test this functionality a bit more easily, maybe you could pass a writer to this function which the formatted output could be written to rather than trying capture what is on stdout.

Reply:
Yea, I could do that. I opted to just use `Example...` tests which tests stdout output automatically. I lost the ability to do a table driven test, but for just testing formatting of a CLI command I think that is OK.